### PR TITLE
FEXServer: Adds new FEXServer service

### PR DESCRIPTION
**Pull request by Sonicadvance1**
This is a relatively invasive change since multiple things needed to
happen at once.

* Socket based logging is removed
  * Logging has been replaced to only support stdout, stderr, and server
  * Server is now default and replaces what FEXLogServer did
  * Server logging now uses a pipe instead of a socket
  * Can be faster than stderr and stdout since the application doesn't
    need to wait on terminal output

* FEXMountDaemon has been removed
  * Functionality has been merged in to FEXServer

* FEXServer is always executed on FEX initialization time
  * Similar in behaviour to Wine's wineserver
  * Can explicitly start this before using FEX for logging purposes
  * Stays around until all instances of FEX exit
  * Will stick around for a short amount of time in case of spurious
    execution

* FEXServer will soon be extended to do more than logging and squashfs
  mounting

* FEX rootfs scripts will need to be updated to support this path
  * Just means rbinding the /tmp folder and forcing a FEXServer instance
    to be alive
* Pressure-vessel works fine in this case since FEXServer will already
  be running
  * It already rbinds the host /tmp folder which is why this works

### CODE REVIEW

File: External/FEXCore/Source/Interface/Config/Config.json.in [hunk 1]
```diff
@@ -219,22 +219,13 @@
...
-      },
       "OutputLog": {
         "Type": "str",
-        "Default": "stderr",
+        "Default": "server",
         "ShortArg": "o",
         "Desc": [
           "File to write FEX output to.",
-          "[stdout, stderr, <Filename>]"
+          "[stdout, stderr, server, <Filename>]"
```
Comment (skmp) [inline External/FEXCore/Source/Interface/Config/Config.json.in:228]: isn't `<Filename>` dropped?

File: Source/Tests/FEXBash.cpp [hunk 2]
```diff
@@ -33,7 +33,7 @@ int main(int argc, char **argv, char **const envp) {
   auto Args = FEX::ArgLoader::Get();
 
   // Ensure RootFS is setup before config options try to pull CONFIG_ROOTFS
-  if (!FEX::RootFS::Setup(envp)) {
+  if (!FEXServerClient::SetupClient(argv[0])) {
     fprintf(stderr, "RootFS configuration failed.");
```
Comment (skmp) [inline Source/Tests/FEXBash.cpp:37]: Shouldn't this be updated from "RootFS ..."?

File: Source/Tests/FEXLoader.cpp [hunk 3]
```diff
@@ -223,7 +233,7 @@ int main(int argc, char **argv, char **const envp) {
   }
 
   // Ensure RootFS is setup before config options try to pull CONFIG_ROOTFS
-  if (!FEX::RootFS::Setup(envp)) {
+  if (!FEXServerClient::SetupClient(argv[0])) {
     LogMan::Msg::EFmt("RootFS failure");
```
Comment (skmp) [inline Source/Tests/FEXLoader.cpp:237]: Shouldn't this be updated from "RootFS ..."?

**Comment by Sonicadvance1** [comment]
https://github.com/FEX-Emu/RootFS has been updated with the new scripts to handle FEXServer changed.
Next release I will rebuild the rootfs images to include it

File: Source/Tools/FEXServer/Main.cpp [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+    // Ignore SIGPIPE, we will be checking for pipe closure which could send this signal
+    signal(SIGPIPE, SIG_IGN);
+    // SIGCHLD if squashfuse exits early.
+    // Ignore it for now
+    signal(SIGCHLD, SIG_IGN);
+  }
+
+  /**
+   * @brief Deparents itself by forking and terminating the parent thread
```
Comment (skmp) [inline Source/Tools/FEXServer/Main.cpp:108]: `thread` -> `process`?

**Review by skmp — APPROVED** [review]
Looks alright.

A bit odd the FEXServer codebase is not class-based, but we can easily change that in a follow up if we want.

File: Source/Tools/FEXServer/Main.cpp (hunk 4 above)
Comment (Sonicadvance1) [reply Source/Tools/FEXServer/Main.cpp]: Resolved.

File: Source/Tests/FEXLoader.cpp (hunk 3 above)
Comment (Sonicadvance1) [reply Source/Tests/FEXLoader.cpp]: Resolved.

File: Source/Tests/FEXBash.cpp (hunk 2 above)
Comment (Sonicadvance1) [reply Source/Tests/FEXBash.cpp]: Resolved.

**Pull request merged**